test_util/src/lsp.rs

```diff
@@ -686,21 +670,70 @@ impl LspClient {
   ) {
     self.initialize_with_config(
       do_build,
-      json!({"deno":{
-        "enable": true
-      }}),
+      json!({ "deno": {
+        "enableBuiltinCommands": true,
+        "enable": true,
+        "cache": null,
+        "certificateStores": null,
+        "codeLens": {
+          "implementations": true,
+          "references": true,
+          "test": true,
+        },
+        "config": null,
+        "importMap": null,
+        "lint": true,
+        "suggest": {
+          "autoImports": true,
+          "completeFunctionCalls": false,
+          "names": true,
+          "paths": true,
+          "imports": {
+            "hosts": {},
+          },
+        },
+        "testing": {
+          "args": [
+            "--allow-all"
+          ],
+          "enable": true,
+        },
+        "tlsCertificate": null,
+        "unsafelyIgnoreCertificateErrors": null,
+        "unstable": false,
+      } }),
     )
   }
 
   pub fn initialize_with_config(
     &mut self,
     do_build: impl Fn(&mut InitializeParamsBuilder),
-    config: Value,
+    mut config: Value,
   ) {
-    let mut builder = InitializeParamsBuilder::new();
+    let mut builder = InitializeParamsBuilder::new(config.clone());
     builder.set_root_uri(self.context.temp_dir().uri());
     do_build(&mut builder);
     let params: InitializeParams = builder.build();
+    // `config` must be updated to account for the builder changes.
+    // TODO(nayeemrmn): Maybe remove builder.
```

Comment:
Why maybe remove the builder?

Reply:
There's too many ways of supplying config, there's a separate arg for that and the value has to be saved to keep supplying it on configuration requests. The builder makes it complicated

Reply:
Oh right, yeah not the whole builder but just make the `InitializeParams::initializationOptions` part of it (which accepts workspace settings) into one value setter.